sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -205,6 +205,29 @@ object DateTimeUtils {
     (day.toInt, MICROSECONDS.toNanos(micros))
   }
 
+  /**
+   *Returns Gregorian days since epoch
+   */
+  def toGregorianDays(
+      year: Int,
+      month: Byte = 1,
+      day: Byte = 1,
+      hour: Byte = 0,
+      minute: Byte = 0,
+      sec: Byte = 0): SQLDate = {
+    val calendar = new GregorianCalendar(year, month - 1, day, hour, minute, sec)
```

Comment:
it is not pure Gregorian  calendar. For 1000-02-29, it is Julian calendar.

Reply:
Ah, okay, should we close SPARK-31212 then? Is it the right behavior to not able to cast `1000-02-29`?

Reply:
The ticket is opened for Spark 2.4 where it is correct date but your PR is for master

Reply:
I see. Didn't know that the semantic has changed in these two versions. Thanks for the explanation. So if I want to fix 2.4, which branch should I create a PR for? branch-2.4?

Reply:
if we can land a minimised fix, let's go ahead for branch-2.4. Otherwise, I think I would rather treat that calendar switching supersedes it.